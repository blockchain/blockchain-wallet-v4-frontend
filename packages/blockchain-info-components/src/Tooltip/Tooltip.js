import React from 'react'
import styled from 'styled-components'
import { transparentize } from 'polished'
import { TextGroup } from '../Text'

const TooltipWrapper = styled.div`
  width: ${props => props.width === 'auto' ? 'auto' : '22px'};
  display: inline-flex;
  position: relative;
`
const TooltipIcon = styled.div`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 16px;
  height: 16px;
  border-radius: 10px;
  background-color: ${props => props.theme[props.colors.backgroundColor]};
  color: ${props => props.theme[props.colors.foreColor]};
  border: 1px solid ${props => props.theme[props.colors.borderColor]};
  cursor: pointer;
  font-size: 12px;
  font-weight: 300;
`
const TooltipLabel = styled.div`
  
`
const TooltipBox = styled(TextGroup)`
  position: absolute;
  bottom: 150%;
  left: -115px;
  text-align: initial;
  width: ${props => props.width === 'auto' || '250px'};
  display: ${props => props.displayed ? 'block' : 'none'};
  background-color: ${props => props.theme['white-blue']};
  color: ${props => props.theme['gray-5']};
  border: 1px solid ${props => props.theme['gray-2']};
  border-radius: 5px;
  padding: 10px 10px;
  box-sizing: border-box;
  box-shadow: ${props => transparentize(0.65, props.theme['gray-6'])} 0px 3px 8px 0px;
  cursor: pointer;
  font-size: 11px;
  font-weight: 300;
  font-family: "Montserrat", sans serif;
  text-align: left;

  > div,
  > span {
    margin-bottom: 0px;
  }

  &:before {
    content: '';
    display: block;
    position: absolute;
    left: 115px;
    top: 100%;
    width: 0;
    height: 0;
    border: 10px solid transparent;
    border-top-color: ${props => props.theme['gray-2']};
  }

  &:after {
    content: '';
    display: block;
    position: absolute;
    left: 116px;
    top: 100%;
    width: 0;
    height: 0;
    border: 9px solid transparent;
    border-top-color: ${props => props.theme['gray-1']};
  }
`

const selectColors = displayed => displayed
  ? { foreColor: 'white', backgroundColor: 'brand-primary', borderColor: 'brand-primary' }
  : { foreColor: 'gray-5', backgroundColor: 'white', borderColor: 'gray-2' }

class Tooltip extends React.PureComponent {
  constructor (props) {
    super(props)
    this.state = { displayed: false }
    this.handleClick = this.handleClick.bind(this)
    this.handleMouseEnter = this.handleMouseEnter.bind(this)
    this.handleMouseLeave = this.handleMouseLeave.bind(this)
  }

  handleClick () {
    this.setState({ displayed: !this.state.displayed })
  }

  handleMouseEnter () {
    this.props.hover && this.setState({ displayed: true })
  }

  handleMouseLeave () {
    this.props.hover && this.setState({ displayed: false })
  }

  render () {
    const icon = this.state.displayed ? 'X' : '?'
    const colors = selectColors(this.state.displayed)

    return (
      <TooltipWrapper width={this.props.width}>
        {
          this.props.label
            ? <TooltipLabel onMouseEnter={this.handleMouseEnter} onMouseLeave={this.handleMouseLeave}>{this.props.label}</TooltipLabel>
            : <TooltipIcon displayed={this.state.displayed} colors={colors} onMouseEnter={this.handleMouseEnter} onMouseLeave={this.handleMouseLeave} onClick={this.handleClick}>{icon}</TooltipIcon>
        }
        <TooltipBox width={this.props.width} displayed={this.state.displayed} onClick={this.handleClick}>{this.props.children}</TooltipBox>
      </TooltipWrapper>
    )
  }
}

export default Tooltip
