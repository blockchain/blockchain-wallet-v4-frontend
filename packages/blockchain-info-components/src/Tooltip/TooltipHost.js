import React from 'react'
import ReactTooltip from 'react-tooltip'
import styled from 'styled-components'

const Wrapper = styled.div`
  width: fit-content;
  display: inline-flex;
`

class TooltipHost extends React.PureComponent {
  componentDidMount () {
    ReactTooltip.rebuild()
  }

  render () {
    const { children, id, tip } = this.props
    const dataTip = tip || []

    return (
      <Wrapper data-tip={dataTip} data-for={id} {...this.props}>
        {children}
      </Wrapper>
    )
  }
}

export default TooltipHost
