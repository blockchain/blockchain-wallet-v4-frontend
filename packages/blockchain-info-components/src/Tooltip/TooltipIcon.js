import styled from 'styled-components'

import { Icon } from '../Icons'

const TooltipIcon = styled(Icon)`
  display: inline-flex;
  margin: 0px 5px;
  font-size: ${props => (props.size ? props.size : '16px')};
`

export default TooltipIcon
