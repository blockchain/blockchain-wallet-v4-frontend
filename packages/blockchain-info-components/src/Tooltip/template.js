import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Text, TextGroup } from '../Text'

const TooltipWrapper = styled.div`
  width: ${props => props.width === 'auto' ? 'auto' : '22px'};
  display: inline-flex;
  position: relative;
`
const TooltipIcon = styled.div`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 16px;
  height: 16px;
  font-size: 12px;
  font-weight: 300;
  > span {
    cursor: pointer;
    &:before {
      color: ${props => props.displayed && props.theme['brand-primary']};
    }
  }
`
const TooltipLabel = styled(Text)`
  font-size: 13px;
  font-weight: 300;
`
const TooltipBox = styled(TextGroup)`
  position: absolute;
  bottom: 150%;
  left: -115px;
  text-align: initial;
  width: ${props => props.width === 'auto' || '250px'};
  display: ${props => props.displayed ? 'block' : 'none'};
  background-color: ${props => props.theme['white-blue']};
  color: ${props => props.theme['gray-5']};
  border: 1px solid ${props => props.theme['gray-2']};
  border-radius: 5px;
  padding: 10px 10px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 11px;
  font-weight: 300;
  font-family: "Montserrat", sans serif;
  text-align: left;

  > div,
  > span {
    margin-bottom: 0px;
  }

  &:before {
    content: '';
    display: block;
    position: absolute;
    left: 115px;
    top: 100%;
    width: 0;
    height: 0;
    border: 10px solid transparent;
    border-top-color: ${props => props.theme['gray-2']};
  }

  &:after {
    content: '';
    display: block;
    position: absolute;
    left: 116px;
    top: 100%;
    width: 0;
    height: 0;
    border: 9px solid transparent;
    border-top-color: ${props => props.theme['white-blue']};
  }
`

export const Tooltip = props => {
  const { icon, width, label, displayed, handleClick, handleMouseEnter, handleMouseLeave } = props
  return (
    <TooltipWrapper width={width}>
      {
        label
          ? <TooltipLabel onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>{label}</TooltipLabel>
          : <TooltipIcon displayed={displayed} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} onClick={handleClick}>{icon}</TooltipIcon>
      }
      <TooltipBox width={width} displayed={displayed} onClick={handleClick}>{props.children}</TooltipBox>
    </TooltipWrapper>
  )
}

Tooltip.propTypes = {
  icon: PropTypes.string,
  width: PropTypes.string,
  label: PropTypes.string,
  displayed: PropTypes.bool,
  handleClick: PropTypes.func,
  handleMouseEnter: PropTypes.func,
  handleMouseLeave: PropTypes.func
}

export default Tooltip
