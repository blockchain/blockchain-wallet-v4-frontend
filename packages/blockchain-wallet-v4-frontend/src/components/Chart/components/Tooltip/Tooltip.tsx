import React from 'react'
import { defaultStyles, TooltipWithBounds as VisxTooltipWithBounds } from '@visx/tooltip'

import { TooltipCard } from './styles'
import { TooltipComponent } from './types'

export const Tooltip: TooltipComponent = ({ children, left, offsetLeft, offsetTop, top }) => {
  return (
    <VisxTooltipWithBounds
      top={top}
      left={left}
      offsetLeft={offsetLeft}
      offsetTop={offsetTop}
      style={{
        ...defaultStyles,
        boxShadow: 'none',
        padding: 'none'
      }}
    >
      <TooltipCard>{children}</TooltipCard>
    </VisxTooltipWithBounds>
  )
}
