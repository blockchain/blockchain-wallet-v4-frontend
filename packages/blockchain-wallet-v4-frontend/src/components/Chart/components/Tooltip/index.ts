export { Tooltip } from './Tooltip'
export type { TooltipComponent, TooltipProps } from './types'
