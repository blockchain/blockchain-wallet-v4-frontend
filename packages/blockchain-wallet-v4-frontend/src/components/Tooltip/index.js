import MenuTooltip from './MenuTooltip.js'

export { MenuTooltip }
