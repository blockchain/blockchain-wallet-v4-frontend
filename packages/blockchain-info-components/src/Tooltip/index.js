import React from 'react'
import onClickOutside from 'react-onclickoutside'
import Tooltip from './template.js'
import { Icon } from '../Icons'

class TooltipContainer extends React.PureComponent {
  constructor (props) {
    super(props)
    this.state = { displayed: false }
    this.handleClick = this.handleClick.bind(this)
    this.handleMouseEnter = this.handleMouseEnter.bind(this)
    this.handleMouseLeave = this.handleMouseLeave.bind(this)
  }

  handleClick () {
    this.setState({ displayed: !this.state.displayed })
  }

  handleMouseEnter () {
    this.props.hover && this.setState({ displayed: true })
  }

  handleMouseLeave () {
    this.props.hover && this.setState({ displayed: false })
  }

  handleClickOutside () {
    this.setState({ displayed: false })
  }

  render () {
    const icon = this.state.displayed ? <Icon name='close-in-circle' /> : <Icon name='question-in-circle' />

    return (
      <Tooltip
        icon={icon}
        width={this.props.width}
        label={this.props.label}
        displayed={this.state.displayed}
        handleClick={this.handleClick}
        handleMouseEnter={this.handleMouseEnter}
        handleMouseLeave={this.handleMouseLeave}>
        {this.props.children}
      </Tooltip>
    )
  }
}

export default onClickOutside(TooltipContainer)
