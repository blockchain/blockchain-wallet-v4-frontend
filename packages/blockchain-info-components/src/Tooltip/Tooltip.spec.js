import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Tooltip from './template'

describe('Tooltip component', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Tooltip id='tip'>
        <span>Tool Text</span>
      </Tooltip>
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
