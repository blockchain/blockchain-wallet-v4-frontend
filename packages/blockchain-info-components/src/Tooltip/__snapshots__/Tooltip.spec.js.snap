// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly 1`] = `
.c3 > * {
  white-space: normal;
}

.c3 > * {
  display: block;
  margin-bottom: 10px;
}

.c0 {
  width: 22px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 16px;
  height: 16px;
  font-size: 12px;
  font-weight: 300;
}

.c1 > span {
  cursor: pointer;
}

.c2 {
  position: absolute;
  bottom: 150%;
  left: -115px;
  text-align: initial;
  width: 250px;
  display: none;
  border: 1px solid;
  border-radius: 5px;
  padding: 10px 10px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 11px;
  font-weight: 300;
  font-family: "Montserrat",sans serif;
  text-align: left;
}

.c2 > div,
.c2 > span {
  margin-bottom: 0px;
}

.c2:before {
  content: '';
  display: block;
  position: absolute;
  left: 115px;
  top: 100%;
  width: 0;
  height: 0;
  border: 10px solid transparent;
}

.c2:after {
  content: '';
  display: block;
  position: absolute;
  left: 116px;
  top: 100%;
  width: 0;
  height: 0;
  border: 9px solid transparent;
}

<div
  className="c0"
  width={undefined}
>
  <div
    className="c1"
    onClick={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
  />
  <div
    className="c2 c3"
    onClick={undefined}
    width={undefined}
  >
    <span>
      Default
    </span>
  </div>
</div>
`;
